src/hooks/core/push-notification/usePushNotificationHook.ts

```diff
@@ -129,6 +129,12 @@ const usePushNotificationHook = () => {
   };
 
   const __handleNotification = async (notification) => {
+    if (notification.data.type === 'topic' && notification?.data?.topic_name) {
+      navigation.navigate('TopicPageScreen', {
+        id: notification?.data?.topic_name
+      });
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code checks if `notification.data.type` is 'topic' and if `notification.data.topic_name` exists. However, it doesn't handle the case where `notification.data` might be undefined, which could lead to a potential TypeError. To avoid this, you should add an additional check for `notification.data`.

```diff
-    if (notification.data.type === 'topic' && notification?.data?.topic_name) {
+    if (notification?.data?.type === 'topic' && notification?.data?.topic_name) {
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->